### PR TITLE
chore: add ci workflow

### PR DESCRIPTION
**Pull request by philjhale**
## Description

* Added CI workflow based on [this template](https://github.com/videojs/.github/blob/main/workflow-templates/ci.yml). I've updated some package versions where it was complaining about deprecated Node.js 12 actions.
* Updated package-lock.json as it was out of date.

### CODE REVIEW

**Comment by gkatsev** [comment]
What about using the ci template from https://github.com/videojs/.github/blob/main/workflow-templates/ci.yml?

**Review by misteroneill — APPROVED** [review]

**Comment by misteroneill** [comment]
Gary makes a good point though!

**Comment by philjhale** [comment]
> What about using the ci template from https://github.com/videojs/.github/blob/main/workflow-templates/ci.yml?

Thanks. I looked at that workflow but wasn't sure if we needed all those steps for a plugin. If we do, it's no problem, I'll update the workflow. 

I have a few questions, just for my understanding:
* Does the videojs workflow run tests on Browserstack rather than locally? If so are the secrets available for all videojs repos? E.g. `BROWSER_STACK_USERNAME`. 
* Are the `ffmpeg/pulseaudio` steps required for this repo?
* Why does the template use `npm i --prefer-offline --no-audit` rather than `npm ci`? My understanding was that `npm ci` was designed for use in CI.

Thanks!

**Comment by gkatsev** [comment]
Yup, the browserstack credentials are available at the org level.
The ffmpeg step may not be stricly necessary, but keeping the action the same across repos is definitely helpful.
I'm not sure exactly why it isn't `npm ci`. My guess is that it might end up being faster due to the caching compared to using `npm ci`.

**Comment by philjhale** [comment]
I've updated the workflow based on the video.js template.

**Comment by philjhale** [comment]
Could someone check with actions are enabled for this repo please? Under Settings->Actions.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        node-version: '${{steps.nvm.outputs.NVMRC}}'
+        cache: npm
+
+    # turn off the default setup-node problem watchers...
+    - run: echo "::remove-matcher owner=eslint-compact::"
+    - run: echo "::remove-matcher owner=eslint-stylish::"
+    - run: echo "::remove-matcher owner=tsc::"
+
+    - name: npm install
+      run: npm ci
```
Comment (gkatsev) [inline .github/workflows/ci.yml:61]: I would keep the --no-audit flag

Comment (philjhale) [reply .github/workflows/ci.yml]: Good call. That's updated.

**Pull request closed without merging**